frontend/test/pytest/temp_env.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Temporarily modify environment variables.
+"""
+import os
+
+# pylint: disable=missing-class-docstring
+class TempEnv:
+
+    def __init__(self, **kwargs):
+        self.backup = os.environ
+        self.kwargs = kwargs
+
+    def __enter__(self):
+        for key, value in self.kwargs.items():
+            os.environ[key] = value
+
+    def __exit__(self, exc_t, exc_v, exc_tb):
+        for key in self.kwargs:
+            del os.environ[key]
+        for key, value in self.backup.items():
+            os.environ[key] = value
```

Comment:
@erick-xanadu no need to write this ourselves, Pytest has a built-in fixture for temporarily changing env variables in tests :)

Reply:
https://docs.pytest.org/en/latest/reference/fixtures.html#reference-fixtures

Reply:
The fixture you want is `monkeypatch`, which has a `setenv()` method.

Reply:
Thanks @josh146, changed [here](https://github.com/PennyLaneAI/catalyst/pull/30/commits/efbfb7d98c3cf47f3b57489ce16126d24cca92e4)

Reply:
Hi @josh146 , I do use `pytest.warns` to check that warnings are raised. But I use:

```python
        with warnings.catch_warnings():
            warnings.simplefilter("error")
```

to make sure that warnings are not raised. If this is the boilerplate you mean, I do not think this is boilerplate. If this is just a note, that is fine. Thanks!

Reply:
Oops, apologies @erick-xanadu, I think I misread your code!